src/js/components/Shared/TracksList.react.js

```diff
@@ -189,158 +124,224 @@ export default class TracksList extends Component {
+                        <div className='cell cell-genre'>
+                            { track.genre.join(', ') }
+                        </div>
+                    </TrackRow>
+                );
+            });
+
+            const tracksListTileStyles = {
+                transform: `translate3d(0, ${(((tilesScrolled * this.rowHeight * chunkLength) + (indexChunk * this.rowHeight * chunkLength)))}px, 0)`
+            };
+
+            return (
+                <div className='tracks-list-tile' key={ indexChunk } style={ tracksListTileStyles }>
+                    { list }
+                </div>
+            );
+        });
     }
 
-    onKey(e) {
+    _toggleSelectionById(id) {
+        let selected = [...this.state.selected];
 
-        const selected = this.state.selected,
-            tracks     = this.props.tracks;
+        if(selected.includes(id)) {
+            // remove track
+            selected.splice(selected.indexOf(id), 1);
 
-        let i = 0;
+        } else {
+            // add track
+            selected.push(id);
+        }
 
-        switch(e.keyCode) { // Todo: fix when node not in dom
-            case 38: // up
+        selected = utils.simpleSort(selected, 'asc');
+        this.setState({ selected });
+    }
 
-                for(let length = tracks.length; i < length; i ++) {
-                    if(selected.includes(tracks[i]._id)) break;
-                }
+    _multiSelect(e, id, index) {
+        const self   = this;
+        const tracks = this.props.tracks;
+        const selected = this.state.selected;
 
-                if(i - 1 >= 0) {
-                    this.setState({ selected : tracks[i - 1]._id }, () => {
+        switch(selected.length) {
+            case 0: {
+                selected.push(id);
+                this.setState({ selected });
+                break;
+            }
+            case 1: {
 
-                        const container = document.querySelector('.tracks-list-container .tracks-list-body');
-                        const nodeOffsetTop = (i - 1) * this.rowHeight;
+                const onlySelected = selected[0];
+                let onlySelectedIndex;
 
-                        if(container.scrollTop > nodeOffsetTop) container.scrollTop = nodeOffsetTop;
-                    });
+                for(let i = 0, length = tracks.length; i < length; i++) { // STH wrong here
+                    if(tracks[i]._id === onlySelected) {
+                        onlySelectedIndex = i;
+                        break;
+                    }
                 }
-                break;
 
-            case 40: // down
+                if(index < onlySelectedIndex) {
+                    for(let i = 1; i <= Math.abs(index - onlySelectedIndex); i++) {
+                        selected.push(tracks[onlySelectedIndex - i]._id);
+                    }
+                } else if(index > onlySelectedIndex) {
+                    for(let i = 1; i <= Math.abs(index - onlySelectedIndex); i++) {
```

Comment:
Yea, we probably could...I'm more inclined to delete all of `case 0` and `case 1`, because they're not necessary.